### PR TITLE
[microNPU] Add MobileNetv2 test to network tests

### PR DESCRIPTION
**Pull request by jacobbohlin**
This adds MobileNetv2 tflite testcase to the microNPU's network tests.

Co-authored-by: Jacob Bohlin <jacob.bohlin@arm.com>

### CODE REVIEW

**Comment by jacobbohlin** [comment]
@manupa-arm @mbaret @lhutton1 @ekalda @NicolaLancellotti @dchauhan-arm

**Review by lhutton1 — COMMENTED** [review]
Thanks @jacobbohlin! I left a few comments, let me know what you think

File: tests/python/contrib/test_ethosu/test_networks.py [hunk 1]
```diff
@@ -65,5 +65,31 @@ def test_forward_mobilenet_v1(accel_type):
...
+    mod = partition_for_ethosu(relay_mod, params)
+    compiled_models = infra.build_source(
+        mod, input_data, output_data, accel_type, output_tolerance=10
+    )
+    infra.verify_source(compiled_models, accel_type)
+
+
 if __name__ == "__main__":
-    test_forward_mobilenet_v1()
+    test_forward_mobilenet_v1(ACCEL_TYPES[0])
```
[the remark this answers is not in the document]
Comment (jacobbohlin) [reply tests/python/contrib/test_ethosu/test_networks.py:95]: Agreed, that's probably better.

File: tests/python/contrib/test_ethosu/test_networks.py [hunk 2]
```diff
@@ -65,5 +65,31 @@ def test_forward_mobilenet_v1(accel_type):
...
+    input_shape = (1, 224, 224, 3)
+    in_min, in_max = util.get_range_for_dtype_str(input_dtype)
+    input_data = np.random.randint(in_min, high=in_max, size=input_shape, dtype=input_dtype)
+    relay_mod, params = convert_to_relay(tflite_model_buf)
+    input_data = {input_tensor: input_data}
+    output_data = generate_ref_data(relay_mod, input_data)
+
+    mod = partition_for_ethosu(relay_mod, params)
+    compiled_models = infra.build_source(
+        mod, input_data, output_data, accel_type, output_tolerance=10
```
[the remark this answers is not in the document]
Comment (jacobbohlin) [reply tests/python/contrib/test_ethosu/test_networks.py:89]: I can give that a try.

File: tests/python/contrib/test_ethosu/test_networks.py [hunk 3]
```diff
@@ -65,5 +65,31 @@ def test_forward_mobilenet_v1(accel_type):
     infra.verify_source(compiled_models, accel_type)
 
 
+def test_forward_mobilenet_v2(accel_type="ethos-u55-256"):
```
[the remark this answers is not in the document]
Comment (jacobbohlin) [reply tests/python/contrib/test_ethosu/test_networks.py:68]: I cannot say I have a definitive answer for this. Probably not the most efficient way to improve test coverage relative to test time.

**Comment by manupa-arm** [comment]
Thanks @jacobbohlin 

I think we could refactor the tests here to be a single one with stimulus being the model url and accelerator.
Moreover, lets write out a tuples of model url and accelerator variants for each tests -- therefore we have more control/visibility of what is being tested as opposed default parametric explosion of the test space.

Some references that we have done this at other places, also feel free to migrate any utility that you feel we can re-use to aot_test_utils

https://github.com/apache/tvm/blob/0009a308d82a321f2399923ab14b4c088461b4f2/tests/python/relay/aot/test_crt_aot_usmp.py#L205-L219

https://github.com/apache/tvm/blob/0009a308d82a321f2399923ab14b4c088461b4f2/tests/python/relay/aot/test_crt_aot_usmp.py#L232-L240

**Comment by jacobbohlin** [comment]
Thanks @manupa-arm. That is a much better approach, I will update according to your suggestions.

**Review by lhutton1 — APPROVED** [review]
Thanks @jacobbohlin LGTM!

**Comment by jacobbohlin** [comment]
Closing this since it has been superseded by https://github.com/apache/tvm/pull/10362.

**Pull request closed without merging**